server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -154,9 +166,74 @@ public RepositoriesService(
             threadPool.relativeTimeInMillisSupplier()
         );
         this.preRestoreChecks = preRestoreChecks;
+        this.defaultRepository = DEFAULT_REPOSITORY_SETTING.get(settings);
+        clusterService.getClusterSettings()
+            .addSettingsUpdateConsumer(DEFAULT_REPOSITORY_SETTING, this::setDefaultRepository, this::validateDefaultRepository);
         snapshotMetrics.createSnapshotShardsInProgressMetric(this::getShardSnapshotsInProgress);
     }
 
+    /**
+     * Gets the configured default repository.
+     *
+     * @return the default repository name, or an empty string if not configured
+     */
+    public String getDefaultRepository() {
+        return defaultRepository;
+    }
+
+    /**
+     * Sets the default repository value.
+     * This is called by the settings update consumer when the setting changes.
+     *
+     * @param repositoryName the new default repository name
+     */
+    private void setDefaultRepository(String repositoryName) {
+        this.defaultRepository = repositoryName;
+        logger.info("Default repository set to [{}]", repositoryName);
+    }
+
+    /**
+     * Validates that the default repository is registered.
+     * This validator is called when the default repository setting is updated.
+     * Empty strings are allowed to clear the default repository setting.
+     *
+     * @param repositoryName the repository name to validate
+     * @throws IllegalArgumentException if the repository name is not empty and the repository doesn't exist
+     */
+    private void validateDefaultRepository(String repositoryName) {
+        if (Strings.isEmpty(repositoryName)) {
+            logger.info("Default repository cleared");
+            return;
+        }
+        if (isRepositoryRegistered(repositoryName) == false) {
+            throw new IllegalArgumentException(
+                "Repository ["
+                    + repositoryName
+                    + "] is not registered. "
+                    + "Cannot set as default repository. Please register the repository prior to setting as default using PUT /_snapshot/"
+                    + repositoryName
+            );
+        }
+    }
+
+    /**
+     * Checks if a repository is registered in the default project, any other project, or internal repositories.
+     *
+     * @param repositoryName the repository name to check
+     * @return true if the repository is registered, false otherwise
+     */
+    private boolean isRepositoryRegistered(String repositoryName) {
+        if (repositoryOrNull(ProjectId.DEFAULT, repositoryName) != null) {
+            return true;
+        }
+        boolean found = repositories.values().stream().anyMatch(projectRepos -> projectRepos.containsKey(repositoryName));
+        if (found) {
+            return true;
+        }
+        found = internalRepositories.values().stream().anyMatch(projectRepos -> projectRepos.containsKey(repositoryName));
+        return found;
+    }
```

Comment:
Two comments:
1. We should not check internal repositories since those are used for CCR.
2. We should be explicit about it being the default project and not check with any other project